tests/foreman/api/test_discoveredhost.py

```diff
@@ -25,6 +25,22 @@
 from robottelo.utils.datafactory import valid_data_list
 
 
+@pytest.fixture(scope='module')
+def module_hg(default_org, default_location, module_target_sat):
```

Comment:
I would just want to keep it straight, by not creating host group/discovery_rule/auto_provision happening within different organizations/locations

Reply:
We don't prefer touching default entities while running the tests, and there should not be any issues or differences when using module_org and location fixtures

Reply:
This will be because I should be careful while creating the discovery rule within the module org/location and hitting the auto provision API passing the id/name of the org/loc being specific there as they exist within the same org/location, hence not getting into that hassle  :)